### PR TITLE
[routing-manager] retry sending failed RS messages

### PR DESCRIPTION
**Pull request by wgtdkp**
Current implement will start evaluating on-link prefix before successfully sending
`kMaxRtrSolicitations=3` RS messages. This commit fixes this issue by retrying
sending failed RS messages in `kRtrSolicitationRetryFailDelay=60` seconds.

### CODE REVIEW

**Review by simonlingoogle — APPROVED** [review]
LGTM :+1:

File: src/core/border_router/routing_manager.hpp [hunk 1]
```diff
@@ -198,6 +198,8 @@ class RoutingManager : public InstanceLocator
     static constexpr uint32_t kRtrSolicitationInterval     = 4;   // Interval between RSs. In sec.
     static constexpr uint32_t kMaxRtrSolicitationDelay     = 1;   // Max delay for initial solicitation. In sec.
     static constexpr uint32_t kMaxRoutingPolicyDelay       = 1;   // Max delay for routing policy evaluation. In sec.
+    static constexpr uint32_t kRtrSolicitationRetryFailDelay =
+        60; // The delay before re-trying sending failed Router Solicitations. In Sec.
```
Comment (abtink) [inline src/core/border_router/routing_manager.hpp:202]: How about renaming the constant name and making the comment shorter so it fits in same line (so it looks nicer)?
Something like this maybe:
```suggestion
    static constexpr uint32_t kRtrSolicitationRetryDelay = 60; // Delay to re-try  failed Router Sol. tx. In Sec.
```

**Review by abtink — APPROVED** [review]
LGTM. Thanks 👍 

One small nit suggestion below. Looks ok to me as is.

File: src/core/border_router/routing_manager.hpp (hunk 1 above)
Comment (wgtdkp) [reply src/core/border_router/routing_manager.hpp]: ha, it's not pretty :(
Updated!

**Review by librasungirl — APPROVED** [review]

**Comment by jwhui** [comment]
@wgtdkp , the Border Router test cases seem to be failing frequently. Can you take a look?

**Comment by wgtdkp** [comment]
> @wgtdkp , the Border Router test cases seem to be failing frequently. Can you take a look?

The test was failing because there are chances that sending RS can fail after enabling the infra interface and routing manager will wait for 60 seconds before retrying. The test is fixed by enlarge the wait time before verifications.

File: src/core/border_router/routing_manager.hpp [hunk 2]
```diff
@@ -198,6 +198,7 @@ class RoutingManager : public InstanceLocator
     static constexpr uint32_t kRtrSolicitationInterval     = 4;   // Interval between RSs. In sec.
     static constexpr uint32_t kMaxRtrSolicitationDelay     = 1;   // Max delay for initial solicitation. In sec.
     static constexpr uint32_t kMaxRoutingPolicyDelay       = 1;   // Max delay for routing policy evaluation. In sec.
+    static constexpr uint32_t kRtrSolicitationRetryDelay   = 60;  // The delay before re-trying failed RS tx. In Sec.
```
Comment (jwhui) [inline src/core/border_router/routing_manager.hpp:201]:
```suggestion
    static constexpr uint32_t kRtrSolicitationRetryDelay   = 60;  // The delay before retrying failed RS tx. In Sec.
```

Comment (wgtdkp) [reply src/core/border_router/routing_manager.hpp]: Thanks!

**Review by jwhui — APPROVED** [review]

**Pull request merged**